icu4c/source/i18n/number_formatimpl.cpp

```diff
@@ -207,6 +207,7 @@ NumberFormatterImpl::macrosToMicroGenerator(const MacroProps& macros, bool safe,
         nsLocal.adoptInstead(ns);
     }
     const char* nsName = U_SUCCESS(status) ? ns->getName() : "latn";
+    uprv_strcpy(fMicros.nsName, nsName);
```

Comment:
Please only copy if uprv_strlen(nsName) < UPRV_LENGTHOF(fMicros.nsName).

Assuming these are BCP 47 subtags they are at most 8 chars long, but it seems good to guard against bad data.